src/ca/team2706/vision/trackerboxreloaded/DataDecoder.java

```diff
@@ -0,0 +1,21 @@
+package ca.team2706.vision.trackerboxreloaded;
+
+import java.util.HashMap;
+
+public class DataDecoder {
+	
+	public static HashMap<String, String> decodeMessage(String message){
+		HashMap<String,String> values = new HashMap<String,String>();
+		if(message.contains("#")){
+			String[] data = message.split("#");
+			for(String s : data){
+				values.put(s.split(":")[0], s.split(":")[1]);
+			}
+		}else{
+			if(message.contains(":")){
+				values.put(message.split(":")[0], message.split(":")[1]);
+			}
+		}
+		return values;
+	}
+}
```

Comment:
I like the idea of providing a wrapper class to read data out of the table at the roboRIO end. It means the roboRIO programmers don't need to worry about how we are transmitting it.

I guess the roboRIO programmers will need to manually copy this class into their code. Once they do that, it'll be really hard to change anything, so we need to be 100% sure that we have thought of every possible thing.